### ISSUE TITLE
Proposal: explore.fast.design should map to the latest versions of available NPM packages

### ISSUE THREAD

**Issue by chrisdholt** [microsoft/fast-dna#2540]
Currently, we've been publishing the explore.fast.design site to reference what is in the latest version of master. Initially, I think this made sense given the state of the project. As we look towards supporting the broader web community, I think it makes more sense to have the site represent what someone may get when installing the components package.

**How do we validate and share things currently in master which are coming in the next publish?**
I think this is initially solved with a "staging" site, such as `explore-stage.fast.design` which can be shared out. This will provide an opportunity for folks to see in-process work as well as give a place to validate/test against for regressions/bugs since the previous release. A longer term solution may be to include a method of selecting between master vs latest release. 

**What does this mean for the build process?**
The build process would primarily remain the same - a new version of the site would kick every commit to master. The primary difference is that that build would be passed to the *staging* site. I think the best start would be to manually push the production site at the time of an NPM publish for the components site. This can be automated further, but I don't think we need to wait to solve that.

**Where do we publicize the staging site (current master)?**
Initially, the only place this would live is in the repo readme which would point to both the production site as well as the staging site with a detailed run down of each. We would start by only linking to the prod site on `fast.design`.

**Comment by bheston** [comment]
This makes sense to me. I presume it's in response to showing the version or deploy date, which seems to help with the separation between published and in-process work.

**Comment by awentzel** [comment]
I'm not exactly following how this differs from the current staging/production release process used elsewhere.  How would the approval process work into Staging and then on to Production?  What would be the steps and order of work to be done in both environments?

**Comment by chrisdholt** [comment]
> I'm not exactly following how this differs from the current staging/production release process used elsewhere. How would the approval process work into Staging and then on to Production? What would be the steps and order of work to be done in both environments?

I'm not aware of a staging environment for this currently; when something goes into master, `explore.fast.design` is updated in production. The proposal would be to manually rev as part of the deployment process under the assumption that folks in the OSS community are not pulling "master", but rather a package version which is fixed. This means if it's been 2 weeks since we've published the site could technically be 2 weeks ahead of the currently published package.

**Comment by awentzel** [comment]
The current services and associated pipelines are in use (internal URL -> external DNS) and exist under ADO FAST project.

* fast-explore-stage.azurewebsites.net -> n/a
* fast-explore-vnext.azurewebsites.net -> n/a
* fast-explore.azurewebsites.net -> explore.fast.design

DNS was intentionally not set up for internal staging or testing sites.  Production currently is a manual only process and requires approval any one of three people and this process gets skipped if the approval is the same person as the approver in staging.  Staging updates upon commit to master.  So I believe this is already setup as you're proposing.  please review and then let's discuss any changes you're proposing.

**Comment by chrisdholt** [comment]
I think there may be a lack of clarity, so I'll correlate my proposal to the list above. 

TLDR; I think your last paragraph is what is represented in *other* projects, but not this project, so that's where the proposal is coming from. It's essentially the recommendation I'm making, so if that was the point you're making then feel free to disregard additional details below. I read your last paragraph as though that's the current process here, which it's not - so that's informing the below clarification.

---

What's missing from the above DNS setup is what is visually represented on the explore site in terms of components. Currently, the master branch correlates to the last bullet:

Essentially it looks like this:
> master -> fast-explore.azurewebsites.net -> explore.fast.design

**Change 1**:  The master branch would deploy to a staging site instead of directly to the internal URL which maps to explore.fast.design.
> `master branch => fast-explore-stage.azurewebsites.net`

**Change 2**: Deployment to production is a manual deployment from internal staging URL to the internal prod url which will map to explore.fast.design.
> `fast-explore-stage.azurewebsites.net => fast-explore.azurewebsites.net => explore.fast.design`

**Comment by awentzel** [comment]
Gotcha.  I've modified the existing pipeline to only deploy production manually.  It had been set to trigger deployment after every release.  This should close this proposal and issue.  A new proposal will be created by @chrisdholt to automate the NPM publishing process.
